module/os/linux/zfs/zpl_file.c

```diff
@@ -161,17 +161,60 @@ static int
 zpl_fsync(struct file *filp, loff_t start, loff_t end, int datasync)
 {
 	struct inode *inode = filp->f_mapping->host;
+	znode_t *zp = ITOZ(inode);
+	zfsvfs_t *zfsvfs = ITOZSB(inode);
 	cred_t *cr = CRED();
 	int error;
 	fstrans_cookie_t cookie;
 
+	ZPL_ENTER(zfsvfs);
+	ZPL_VERIFY_ZP(zp);
```

Comment:
Looking at this again, it seems to me there's no need for the `ZPL_*` calls here.  The inode/znode will have a reference held and these atomics can be safely accessed.  Unless you've seen an issue I'm missing why don't we drop these.

Reply:
> Looking at this again, it seems to me there's no need for the ZPL_* calls here. The inode/znode will have a reference held and these atomics can be safely accessed.

fixed, this does make the code simpler

> Unless you've seen an issue I'm missing why don't we drop these.

nope, i basically followed the pattern in other parts of the code and made these calls -  i wasn't sure about the guarantees about the inode validity at this point.